test/smoke-fails/issue-435/issue-435.c

```diff
@@ -1,20 +1,39 @@
 #include <math.h>
 #include <stdlib.h>
 #include <omp.h>
+#include <stdio.h>
 
 
-int main(int argc, char const * argv[]) {
+int simplesum(int argc, char const * argv[]) {
     const int N = 1000;
     int array[N];
     for (int i=0; i<N; i++){
         array[i] = i;
     }
-    #pragma omp target
+
+    #pragma omp target map(tofrom: array[:N])
     {
         #pragma omp parallel for
-        for (int i=-N; i<N; i++){
-            array[i] += array[(int)abs(i)];
+        for (int i=0; i<N; i++) {
+            array[i] += (int)abs(-i);
```

Comment:
Could this be kept with negative `i` and `abs` as an index? Not sure it makes a difference but just to be sure since it was the initial erroring case.

Reply:
It doesn't make any difference but I have made the index of the array and abs as well.
I didn't keep the original example because it has out of bound accesses and race conditions.